toonz/sources/include/tgeometry.h

```diff
@@ -1178,8 +1135,347 @@ inline TAffine inv(const TAffine &a) { return a.inv(); }
+    inline int index() const { return (m_current - m_begin) / 2; }
+    inline int reverseIndex() const {
+      int i = index();
+      return i == 0 ? 0 : size() - i;
+    }
+    inline bool lapped() const { return m_lapped; }
+    inline bool valid() const { return m_prebegin != m_begin; }
+    inline bool isFull() const { return !valid() && m_flip; }
+    inline bool isEmpty() const { return !valid() && !m_flip; }
+
+    inline operator bool() const { return !m_lapped; }
+
+    inline Iterator &operator++() {
+      if (!valid()) {
+        m_lapped = true;
+        return *this;
+      }
+      m_lapped = (m_current == m_end);
+      if (m_lapped)
+        m_current = m_begin;
+      else
+        m_current += 2;
+      return *this;
+    }
+
+    inline Iterator &operator--() {
+      if (!valid()) {
+        m_lapped = true;
+        return *this;
+      }
+      m_lapped = (m_current == m_end);
+      if (m_lapped)
+        m_current = m_end;
+      else
+        m_current -= 2;
+      return *this;
+    }
+
+    inline Iterator &operator+=(int i) {
+      if (i == 0) {
+        m_lapped = isEmpty();
+        return *this;
+      }
+      if (!valid()) {
+        m_lapped = true;
+        return *this;
+      }
+      int ii = index();
+      int s  = size();
+      if (ii + i >= 0 && ii + i < s) {
+        m_current += i * 2;
+        m_lapped = false;
+      } else {
+        m_current = m_begin + ((ii + s + i % s) % s) * 2;
+        m_lapped  = true;
+      }
+      return *this;
+    }
+
+    inline int operator-(const Iterator &i) const {
+      assert(m_flip == i.m_flip && m_begin == i.m_begin && m_end == i.m_end &&
+             m_prebegin == i.m_prebegin);
+      int ii = m_current - i.m_current;
+      return ii < 0 ? ii + size() : ii;
+    }
+
+    inline Iterator operator++() const {
+      Iterator copy(*this);
+      ++(*this);
+      return copy;
+    }
+    inline Iterator operator--() const {
+      Iterator copy(*this);
+      --(*this);
+      return copy;
+    }
+    inline Iterator &operator-=(int i) { return (*this) += -i; }
+    inline Iterator operator+(int i) const {
+      Iterator ii(*this);
+      return ii += i;
+    }
+    inline Iterator operator-(int i) const {
+      Iterator ii(*this);
+      return ii -= i;
+    }
+  };
+
+private:
+  bool m_flip;
+  List m_angles;
+
+  int find(Type a) const;
+  void insert(Type a);
+  bool doAdd(Type a0, Type a1);
+
+public:
+  inline explicit TAngleRangeSet(bool fill = false) : m_flip(fill) {}
+  inline TAngleRangeSet(const TAngleRangeSet &x, bool flip = false)
+      : m_flip(x.isFlipped() != flip), m_angles(x.angles()) {}
+
+  inline const List &angles() const { return m_angles; }
+  inline bool isFlipped() const { return m_flip; }
+  inline bool isEmpty() const { return !m_flip && m_angles.empty(); }
+  inline bool isFull() const { return m_flip && m_angles.empty(); }
+
+  bool contains(Type a) const;
+  bool check() const;
```

Comment:
Jen.kins for Windows build seems to fail with error at this line, saying;

```
c:\users\opentoonz\.jenkins\workspace\opentoonz project pr-auto@2\toonz\sources\include\tgeometry.h(1448): warning C4003: ??? 'check' ??????????????????? [C:\Users\opentoonz\.jenkins\workspace\OpenToonz project PR-auto@2\toonz\build_srv\image\image.vcxproj]
c:\users\opentoonz\.jenkins\workspace\opentoonz project pr-auto@2\toonz\sources\include\tgeometry.h(1448): error C2208: 'const bool': ??????????????????????????? [C:\Users\opentoonz\.jenkins\workspace\OpenToonz project PR-auto@2\toonz\build_srv\image\image.vcxproj]
```

I'm not sure but the term `check` is defined as macro at elsewhere?